From 24b2d528ba326a9f5b45c97a2ec7b3ea2c6f8527 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Tue, 15 Jan 2019 15:12:01 +0200
Subject: [PATCH 2/7] Minor fix in the testsuite

* tests/symlink-long.at: Use m4 to create dirname, instead of
using bash-specific syntax.
---
 .gitignore            |  2 ++
 tests/symlink-long.at | 12 +++++++-----
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/.gitignore b/.gitignore
index fae03de..92e9421 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,5 @@
+/ABOUT-NLS
+/build-aux
 /ABOUT-NLS~
 *.a
 *.o
diff --git a/tests/symlink-long.at b/tests/symlink-long.at
index 53a9fdd..4b39a8c 100644
--- a/tests/symlink-long.at
+++ b/tests/symlink-long.at
@@ -23,14 +23,14 @@
 AT_SETUP([symlink-long])
 AT_KEYWORDS([symlink-long copyout])
 
+m4_pushdef([DIRNAME],
+[m4_if($1,0,,[xxxxxxxxx/DIRNAME(m4_decr($1))])])
+
 AT_CHECK([
 
 # len(dirname) > READBUFSIZE
-dirname=
-for i in {1..52}; do
-    dirname="xxxxxxxxx/$dirname"
-    mkdir "$dirname"
-done
+dirname=DIRNAME(52)
+AS_MKDIR_P($dirname)
 ln -s "$dirname" x || AT_SKIP_TEST
 
 echo x | cpio -o > ar
@@ -43,4 +43,6 @@ test "$list" = "$dirname" && echo success || echo fail
 2 blocks
 ])
 
+m4_popdef([DIRNAME])
+
 AT_CLEANUP
-- 
2.20.1

